### PR TITLE
feat: Add warning styling to Slider

### PR DESCRIPTION
**Pull request by cansuaa**
### Description

Slider is recently released using FormFieldValidationControlProps which with warning validations includes warning API but the component doesn't have the styling for it yet. This PR addresses this problem.

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pages/slider/permutations.page.tsx [hunk 1]
```diff
@@ -17,6 +17,26 @@ const permutations = createPermutations<SliderProps>([
...
+    min: [-10],
+    max: [10],
+    hideFillLine: [false, true],
+    disabled: [false],
+    tickMarks: [false, true],
+    warning: [true],
+    step: [0.5, 1, 5],
+  },
+  {
+    value: [0],
```
Comment (katiegeorge) [inline pages/slider/permutations.page.tsx:31]: It seems like this is doing the same as the above. Can you combine them and add `disabled: [false, true]` to the top one?

File: src/slider/styles.scss [hunk 2]
```diff
@@ -173,10 +181,18 @@
...
 
+  &.warning {
+    background: awsui.$color-text-status-warning;
+  }
+
   &.error-active {
     background-color: awsui.$color-background-slider-error-pressed;
   }
 
+  &.warning-active {
```
Comment (katiegeorge) [inline src/slider/styles.scss:192]: Pressing the slider doesn't seem to be changing the color. It looks like there is supposed to be two different colors for active and non-active warning, is that a bug?

File: pages/slider/permutations.page.tsx (hunk 1 above)
Comment (cansuaa) [reply pages/slider/permutations.page.tsx:31]: The one below tests the warning slider with disabled setting, combining them exceeds the page limitation that's why I separated this part.

**Review by katiegeorge — APPROVED** [review]
Will resolve pressed warning color discussion in a separate PR. Looks great!

**Pull request merged**